services/cd-service/pkg/sqlitestore/sqlitestore.go

```diff
@@ -0,0 +1,59 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+package sqlitestore
+
+// #cgo pkg-config: sqlite3 libgit2
+/*
+#include <git2.h>
+#include <sqlite3.h>
+#include "sqlite.h"
+*/
+import "C"
+import (
+	"fmt"
+	"unsafe"
+
+	git "github.com/libgit2/git2go/v34"
+)
+
+func NewOdbBackend(name string) (*git.OdbBackend, error) {
+	var (
+		result  *C.git_odb_backend
+		err_out *C.char
+	)
+	err := C.kp_backend_sqlite(&result, C.CString(name), &err_out)
+	if err != C.SQLITE_OK {
+		str := C.GoString(C.sqlite3_errstr(err))
+		return nil, fmt.Errorf("sqlitestore: %d %s %s", err, str, C.GoString(err_out))
+	}
+	return git.NewOdbBackendFromC(unsafe.Pointer(result)), nil
+}
+
+func NewOdb(name string) (*git.Odb, error) {
```

Comment:
the `NewOdb` function is a little similar to
https://github.com/freiheit-com/kuberpult/pull/592/files#diff-193ecd1a01c5cb50ce7f6f737973a08c44214da29a83c1467038ed5b64157878R134

Can we combine them?

Reply:
Yes. The method is actually unused. I've deleted it.